### PR TITLE
created adapter pattern

### PR DESCRIPTION
**Pull request by EuhenySiniy**

### CODE REVIEW

File: src/main/java/patterns/adapter/AdapterForMercedes.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package patterns.adapter;
+
+public class AdapterForMercedes extends Mercedes implements Car {
+    public void startEngine() {
```
Comment (Taniann) [inline src/main/java/patterns/adapter/AdapterForMercedes.java]: add @override annotation

File: src/main/java/patterns/adapter/Mercedes.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package patterns.adapter;
+
+public class Mercedes {
+    public void mercedesStartEngine() {
```
Comment (Taniann) [inline src/main/java/patterns/adapter/Mercedes.java]: the same

File: src/main/java/patterns/adapter/AdapterForMercedes.java (hunk 1 above)
Comment (EuhenySiniy) [reply src/main/java/patterns/adapter/AdapterForMercedes.java]: Override annotation is not allowed when implementing interface method. If i not wrong

File: src/main/java/patterns/adapter/Mercedes.java (hunk 2 above)
Comment (EuhenySiniy) [reply src/main/java/patterns/adapter/Mercedes.java]: We need override in superclass?

Comment (EuhenySiniy) [reply src/main/java/patterns/adapter/Mercedes.java]: image

**Comment by EuhenySiniy** [comment]
I'm sorry for file from target folder

File: src/main/java/patterns/adapter/Mercedes.java (hunk 2 above)
Comment (Taniann) [reply src/main/java/patterns/adapter/Mercedes.java]: yes sorry
I didnt notice that this class dont implement interface or extend another class

File: src/main/java/patterns/flyweight/Garage.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+public class Garage {
+    private static final Map<String, Car> cars = new HashMap<String, Car>();
+
+    public Car getCarByName(String brand) {
+        Car car = cars.get(brand);
+
+        if (car == null) {
+            switch (brand) {
+                case "volvo":
```
Comment (Taniann) [inline src/main/java/patterns/flyweight/Garage.java]: better use constants for these purpose
"volvo", "renault"

File: src/main/java/patterns/prototype/Car.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package patterns.prototype;
+
+import lombok.AllArgsConstructor;
+
+@AllArgsConstructor
+public class Car implements Copyable {
+    private String brand;
+    private String model;
+    private int horsePower;
+
```
Comment (Taniann) [inline src/main/java/patterns/prototype/Car.java]: remove redundant blank line

File: src/main/java/patterns/factory/builder/Plants.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
+package patterns.factory.builder;
+
+import patterns.factory.Fruit;
+
+public class Plants implements Fruit {
```
Comment (Taniann) [inline src/main/java/patterns/factory/builder/Plants.java]: add getters

**Pull request merged**